src/algorithms/tracking/TrackPropagation.h

```diff
@@ -44,8 +44,8 @@ namespace eicrecon {
 
         /** Propagates a collection of trajectories to a given surface
          * @remark: being a simple wrapper of propagate(...) this method is more suitable for factories */
-        std::vector<std::unique_ptr<edm4eic::TrackPoint>> propagateMany(std::vector<const ActsExamples::Trajectories *> trajectories,
-                                                         const std::shared_ptr<const Acts::Surface> &targetSurf);
+        std::unique_ptr<edm4eic::TrackSegmentCollection> propagateMany(std::vector<const ActsExamples::Trajectories *> trajectories,
+                                                         const std::vector<std::shared_ptr<Acts::Surface>> &target_surface_list);
```

Comment:
So `propagateMany` now propagates to a list of surfaces, as `propagateToSurfaceList` does? If so, consider combining these methods.

Reply:
That's the idea, yeah. There are two users of this TrackPropagation algorithm right now: RICH projections to radiators and track projections to calorimeters. Aligning the methodology will only facilitate future maintenance (not to mention avoid duplicate evaluations). They differ in how the surfaces are passed, with the RICH using its ActsGeo interface in the RichGeo service and the TrackPropagation_factory doing hard-coded string lookups in the factory. In both cases it's non-ideal how much is done outside the algorithm, in my opinion. But a first step is to have both of these methodologies use the same algorithm interface/entry point, while keeping the current surface construction unchanged.

Reply:
> non-ideal how much is done outside the algorithm

`RichTrack_factory` is also guilty here (which uses `richgeo`'s `ActsGeo`). For the RICHes our preference was to keep anything related to the geometry in `richgeo` to keep subsystem dependence out of the algorithms. At the moment only the dRICH uses `richgeo`, but the doors are open for the pfRICH (and possibly DIRC); the pfRICH already has an (untested) IRT geometry representation.